allauth/socialaccount/adapter.py

```diff
@@ -197,9 +197,22 @@ def get_app(self, request, provider):
 
         config = app_settings.PROVIDERS.get(provider, {}).get("APP")
         if config:
-            app = SocialApp(provider=provider)
-            for field in ["client_id", "secret", "key", "certificate_key"]:
-                setattr(app, field, config.get(field))
+            if (
+                app_settings.STORE_TOKENS
+                and SocialApp.objects.filter(provider=provider).exists()
+            ):
+                # We already saved the app. Fetch and return it right away...
+
+                # FIXME: Ideally there should be a way to specify specific sites
+                # with settings-based credentials. We currently don't, so we can't
+                # use the get_current method that consider sites as well.
+                app = SocialApp.objects.get(provider=provider)
+            else:
+                app = SocialApp(provider=provider)
+                for field in ["client_id", "secret", "key", "certificate_key"]:
+                    setattr(app, field, config.get(field))
```

Comment:
Is it necessary to validate that these values are present here? I think don't want to create an incomplete `SocialApp` object here, then cause other issues later on.